packages/db-states-redis/lib/states/statesInRedisClient.js

```diff
@@ -49,30 +46,28 @@ class StateRedis {
         this.client = null;
         this.sub = null;
         this.subSystem = null;
-        const ioRegExp = new RegExp('^' + this.namespaceRedis.replace(/\./g, '\\.') + '[_A-Za-z0-9ÄÖÜäöüа-яА-Я]+'); // io.[_A-Za-z0-9]+
 
-        this.log = this.settings.logger;
-        if (!this.log) {
-            this.log = {
-                silly: function (_msg) {/* console.log(msg); */
-                },
-                debug: function (_msg) {/* console.log(msg); */
-                },
-                info: function (_msg) {/* console.log(msg); */
-                },
-                warn: function (msg) {
-                    console.log(msg);
-                },
-                error: function (msg) {
-                    console.log(msg);
-                }
-            };
-        } else if (!this.log.silly) {
-            this.log.silly = this.log.debug;
+        this.log = tools.getLogger(this.settings.logger);
+
+        if (this.settings.autoConnect === undefined || this.settings.autoConnect) {
```

Comment:
what type does `autoConnect` have? if `boolean|undefined` I would change this to `if (this.settings.autoConnect !== false)`

Reply:
good idea